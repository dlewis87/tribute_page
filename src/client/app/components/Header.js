import React from 'react';
import { Jumbotron } from 'react-bootstrap';
import Gallery from './Gallery';

const Header = () => (
	<Jumbotron>
		<hr />
		<div className="row">
			<div className="col-xs-12">
				<h1 className="text-center">Ada Lovelace</h1>
				<h4 className="text-center">The first computer programmer</h4>
			</div>
		</div>
		<div className="row">
			<div className="col-xs-12">
        <Gallery />
      </div>
    </div>
  </Jumbotron>
);

export default Header;