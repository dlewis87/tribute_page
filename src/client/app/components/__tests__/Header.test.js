import * as React from 'react';
import { shallow } from 'enzyme';
import Header from '../Header';

describe('Header', () => {
  const header = shallow(<Header />);

  it('renders properly', () => {
    expect(header).toMatchSnapshot();
  });

  it('contains a Jumbotron component', () => {
    expect(header.find('Jumbotron').exists()).toBe(true);
  });

  it('contains a Gallery component', () => {
    expect(header.find('Gallery').exists()).toBe(true);
  });
});