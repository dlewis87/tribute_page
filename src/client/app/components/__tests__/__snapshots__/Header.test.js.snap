// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "bsClass": "jumbotron",
      "children": Array [
        <hr />,
        <div
          className="row"
        >
          <div
            className="col-xs-12"
          >
            <h1
              className="text-center"
            >
              Ada Lovelace
            </h1>
            <h4
              className="text-center"
            >
              The first computer programmer
            </h4>
          </div>
        </div>,
        <div
          className="row"
        >
          <div
            className="col-xs-12"
          >
            <Gallery />
          </div>
        </div>,
      ],
      "componentClass": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col-xs-12"
          >
            <h1
              className="text-center"
            >
              Ada Lovelace
            </h1>
            <h4
              className="text-center"
            >
              The first computer programmer
            </h4>
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="text-center"
              >
                Ada Lovelace
              </h1>,
              <h4
                className="text-center"
              >
                The first computer programmer
              </h4>,
            ],
            "className": "col-xs-12",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Ada Lovelace",
                "className": "text-center",
              },
              "ref": null,
              "rendered": "Ada Lovelace",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "The first computer programmer",
                "className": "text-center",
              },
              "ref": null,
              "rendered": "The first computer programmer",
              "type": "h4",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col-xs-12"
          >
            <Gallery />
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <Gallery />,
            "className": "col-xs-12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "bsClass": "jumbotron",
        "children": Array [
          <hr />,
          <div
            className="row"
          >
            <div
              className="col-xs-12"
            >
              <h1
                className="text-center"
              >
                Ada Lovelace
              </h1>
              <h4
                className="text-center"
              >
                The first computer programmer
              </h4>
            </div>
          </div>,
          <div
            className="row"
          >
            <div
              className="col-xs-12"
            >
              <Gallery />
            </div>
          </div>,
        ],
        "componentClass": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-xs-12"
            >
              <h1
                className="text-center"
              >
                Ada Lovelace
              </h1>
              <h4
                className="text-center"
              >
                The first computer programmer
              </h4>
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="text-center"
                >
                  Ada Lovelace
                </h1>,
                <h4
                  className="text-center"
                >
                  The first computer programmer
                </h4>,
              ],
              "className": "col-xs-12",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Ada Lovelace",
                  "className": "text-center",
                },
                "ref": null,
                "rendered": "Ada Lovelace",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "The first computer programmer",
                  "className": "text-center",
                },
                "ref": null,
                "rendered": "The first computer programmer",
                "type": "h4",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-xs-12"
            >
              <Gallery />
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <Gallery />,
              "className": "col-xs-12",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
